opensrp-giz-malawi/src/test/java/org/smartregister/giz/processor/GizMalawiProcessorForJavaTest.java

```diff
@@ -0,0 +1,216 @@
+        serviceTypeDeworming2.setId(1L);
+        serviceTypeDeworming2.setName("Deworming_2");
+        serviceTypeDeworming2.setPrerequisite("prerequisite|Deworming 1");
+        serviceTypeDeworming2.setPreOffset("+6m");
+        serviceTypeDeworming2.setUnits("200 mg");
+        serviceTypeDeworming2.setDateEntity("encounter");
+
+        ServiceType serviceTypeDeworming3 = new ServiceType();
+        serviceTypeDeworming3.setName("Deworming_3");
+        serviceTypeDeworming3.setPrerequisite("prerequisite|Deworming 2");
+        serviceTypeDeworming3.setPreOffset("+6m");
+        serviceTypeDeworming3.setUnits("400 mg");
+        serviceTypeDeworming3.setDateEntity("encounter");
+
+        Mockito.when(recurringServiceTypeRepository.searchByName("Deworming 1")).thenReturn(Arrays.asList(serviceTypeDeworming2, serviceTypeDeworming3));
+        Mockito.when(processorForJava.processCaseModel(ArgumentMatchers.any(EventClient.class), ArgumentMatchers.any(Table.class))).thenReturn(contentValues);
+        Mockito.when(contentValues.size()).thenReturn(7);
+        Mockito.when(contentValues.getAsString(HeightRepository.ANMID)).thenReturn("provider");
+        Mockito.when(contentValues.getAsString(HeightRepository.LOCATIONID)).thenReturn("lombwe");
+        Mockito.when(contentValues.getAsString(RecurringServiceTypeRepository.NAME)).thenReturn("Deworming_1");
+        Mockito.when(contentValues.getAsString(RecurringServiceRecordRepository.DATE)).thenReturn("2019-09-27 09:45:44");
+
+        Mockito.when(contentValues.getAsString(HeightRepository.CREATED_AT)).thenReturn("2019-09-27 09:45:44");
+
+        Event event = new Event();
+        event.setEventId("231");
+        event.setFormSubmissionId("1234");
+
+        Table table = new Table();
+        table.name = "recurring_service_records";
+        Object[] params = {new EventClient(event, new Client("23")), table};
+        Boolean result = Whitebox.invokeMethod(processorForJava, "processService", params);
+
+        Mockito.verify(recurringServiceRecordRepository).add(recordServiceArgumentCaptor.capture());
+        ServiceRecord resultServiceRecord = recordServiceArgumentCaptor.getValue();
+        Assert.assertEquals(Utils.getDate("2019-09-27 09:45:44"), resultServiceRecord.getDate());
+        Assert.assertEquals(java.util.Optional.of(1L).get(), resultServiceRecord.getRecurringServiceId());
+        Assert.assertEquals("Deworming 1", resultServiceRecord.getName());
+        Assert.assertEquals(RecurringServiceRecordRepository.TYPE_Synced, resultServiceRecord.getSyncStatus());
+
+        Assert.assertEquals("231", resultServiceRecord.getEventId());
+        Assert.assertEquals("1234", resultServiceRecord.getFormSubmissionId());
+        Assert.assertEquals("1234", resultServiceRecord.getFormSubmissionId());
+        Assert.assertTrue(result);
+    }
+
+    @Test
+    public void processBCScarEventWithValidEventClient() throws Exception {
```

Comment:
Specify return type in the test naming **(Better test naming)**

I think **methodActionWithVariablesShouldReturnOrDoX** is the convention